### PR TITLE
Use Path objects for ComponentCreate

### PR DESCRIPTION
**Pull request by mirpedrol**
Follow up from #2549

- Replace the library os by pathlib in ComponentCreate
- Update modules and subworkflows template folder structure

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/module-template/tests/tags.yml [hunk 1]
```diff
@@ -1,2 +1,2 @@
-{{ component_dir }}:
+{ { component_dir } }:
```
Comment (mirpedrol) [inline nf_core/module-template/tests/tags.yml:1]:
```suggestion
{{ component_dir }}:
```

File: nf_core/components/create.py [hunk 2]
```diff
@@ -394,18 +391,12 @@ def _get_component_dirs(self):
 
             # Set file paths
             # For modules - can be tool/ or tool/subtool/ so can't do in template directory structure
-            file_paths[os.path.join(self.component_type, "main.nf")] = os.path.join(component_dir, "main.nf")
-            file_paths[os.path.join(self.component_type, "meta.yml")] = os.path.join(component_dir, "meta.yml")
+            file_paths["main.nf"] = component_dir / "main.nf"
+            file_paths["meta.yml"] = component_dir / "meta.yml"
```
Comment (mashehu) [inline nf_core/components/create.py:395]:
```suggestion
            file_paths["main.nf"] = Path(component_dir,"main.nf")
            file_paths["meta.yml"] = Path(component_dir,"meta.yml")
```

File: nf_core/components/create.py [hunk 3]
```diff
@@ -435,14 +425,12 @@ def _get_username(self):
     def _copy_old_files(self, component_old_path):
         """Copy files from old module to new module"""
         log.debug("Copying original main.nf file")
-        shutil.copyfile(component_old_path / "main.nf", self.file_paths[self.component_type + "/main.nf"])
+        shutil.copyfile(component_old_path / "main.nf", self.file_paths["main.nf"])
```
Comment (mashehu) [inline nf_core/components/create.py:428]:
```suggestion
        shutil.copyfile(Path(component_old_path, "main.nf"), self.file_paths["main.nf"])
```

File: nf_core/components/create.py [hunk 4]
```diff
@@ -435,14 +425,12 @@ def _get_username(self):
...
         log.debug("Copying original meta.yml file")
-        shutil.copyfile(component_old_path / "meta.yml", self.file_paths[self.component_type + "/meta.yml"])
+        shutil.copyfile(component_old_path / "meta.yml", self.file_paths["meta.yml"])
```
Comment (mashehu) [inline nf_core/components/create.py:430]:
```suggestion
        shutil.copyfile(Path(component_old_path, "meta.yml"), self.file_paths["meta.yml"])
```

File: nf_core/components/create.py [hunk 5]
```diff
@@ -435,14 +425,12 @@ def _get_username(self):
...
         if self.component_type == "modules":
             log.debug("Copying original environment.yml file")
-            shutil.copyfile(
-                component_old_path / "environment.yml", self.file_paths[self.component_type + "/environment.yml"]
-            )
+            shutil.copyfile(component_old_path / "environment.yml", self.file_paths["environment.yml"])
```
Comment (mashehu) [inline nf_core/components/create.py:433]:
```suggestion
            shutil.copyfile(Path(component_old_path, "environment.yml"), self.file_paths["environment.yml"])
```

File: nf_core/components/create.py [hunk 6]
```diff
@@ -279,16 +278,16 @@ def _render_template(self):
...
                 log.debug(f"Writing output to: '{dest_fn}'")
                 fh.write(rendered_output)
 
             # Mirror file permissions
-            template_stat = os.stat(
-                os.path.join(os.path.dirname(nf_core.__file__), f"{self.component_type[:-1]}-template", template_fn)
-            )
-            os.chmod(dest_fn, template_stat.st_mode)
+            template_stat = (
+                Path(nf_core.__file__).parent / f"{self.component_type[:-1]}-template" / template_fn
```
Comment (mashehu) [inline nf_core/components/create.py:288]:
```suggestion
                Path(Path(nf_core.__file__).parent, f"{self.component_type[:-1]}-template", template_fn)
```

File: nf_core/components/create.py [hunk 7]
```diff
@@ -339,17 +338,17 @@ def _get_component_dirs(self):
         """
         file_paths = {}
         if self.repo_type == "pipeline":
-            local_component_dir = os.path.join(self.directory, self.component_type, "local")
+            local_component_dir = Path(self.directory, self.component_type, "local")
             # Check whether component file already exists
-            component_file = os.path.join(local_component_dir, f"{self.component_name}.nf")
-            if os.path.exists(component_file) and not self.force_overwrite:
+            component_file = local_component_dir / f"{self.component_name}.nf"
```
Comment (mashehu) [inline nf_core/components/create.py:343]:
```suggestion
            component_file = Path(local_component_dir, f"{self.component_name}.nf")
```

File: nf_core/components/create.py [hunk 8]
```diff
@@ -339,17 +338,17 @@ def _get_component_dirs(self):
...
+            if component_file.exists() and not self.force_overwrite:
                 raise UserWarning(
                     f"{self.component_type[:-1].title()} file exists already: '{component_file}'. Use '--force' to overwrite"
                 )
 
             if self.component_type == "modules":
                 # If a subtool, check if there is a module called the base tool name already
-                if self.subtool and os.path.exists(os.path.join(local_component_dir, f"{self.component}.nf")):
+                if self.subtool and (local_component_dir / f"{self.component}.nf").exists():
```
Comment (mashehu) [inline nf_core/components/create.py:351]:
```suggestion
                if self.subtool and Path(local_component_dir, f"{self.component}.nf").exists():
```

File: nf_core/components/create.py (hunk 2 above)
Comment (mirpedrol) [reply nf_core/components/create.py:395]: I don't think we need to explicitly add `Path()` everywhere, or is there a reason that I am missing?

Comment (mashehu) [reply nf_core/components/create.py:395]: I think I am now just used to overuse Path() 🙂  if you prefer the other way (and it is a bit more readable), it's also totally fine for me (and with type hints we should get an error message, when we try to concat a string with a string)

Comment (mirpedrol) [reply nf_core/components/create.py:395]: I will keep it with `/` then if that's ok 😄

**Pull request merged**